drivers/can/Kconfig

```diff
@@ -193,4 +193,44 @@ config MCP2515_SPI_SCK_FREQUENCY
 
 endif # CAN_MCP2515
 
-endif
+menuconfig CAN_SJA1000
+	tristate "Philips/NXP SJA1000 devices"
+
+if CAN_SJA1000
+
+config CAN_SJA1000_BITRATE
+	int "SJA1000 bitrate"
+	default 1000000
+	---help---
+		SJA1000 bit rate.
+
+config CAN_SJA1000_SJW
+	int "SJA1000 Synchronization Jump Width"
+	default 1
+	range 1 4
+	---help---
+		The duration of a synchronization jump is SJW.
+
+config CAN_SJA1000_SAM
+	bool "The CAN bus is sampled 3 times"
+	default false
+	---help---
+		Recommended for low to medium speed buses
+		to filter spikes on the bus-line.
+
+config CAN_SJA1000_SAMPLEP
+	int "SJA1000 CAN sample point (Percentage)"
+	default 87
+	range 0 100
+
```

Comment:
@radek-pesina I think the sample point can be defined automatically when Propagation Delay, Phase Segment 1, Phase Segment 2 and Sync Jump Width are supplied. Please see MCP2515 driver for reference. Also this site: http://www.bittiming.can-wiki.info/ and mbtime tool: https://web.archive.org/web/20140626031046/http://www.intrepidcs.com/support/mbtime.htm are good source of information to get it working correctly.

Reply:
This is carrying-over the existing TWAI method of calculating bitrate, where instead of defining Propagation Delay, Phase Segment 1, Phase Segment 2 and Sync Jump Width to determine the baud rate, the values of TSEG1 and TSEG 2 are automatically calculated based on SJW, sample point and the requested baud rate.  They both achieve the same function, but one is programatically while the other is a more explicit setting of each parameter (e.g. setting PD, PS1, PS2 and SJW determines the resulting sample point, else by defining the sample point and SJW you can determine PD, PS1 and PS2).  Happy to go either way.  Let me know your thoughts.

Reply:
I think it is up to you! I think Espressif used SJW can percentage to make it easy to setup for end users, but I think the other way is more flexible and will generate realistic sampling point (the other will try to get near to desired sampling point)

Reply:
It is much more portable to define bitrate in Hz and then sample point in percentage. It has some disadvantage that bittiming code has to chose between different possible prescaller values to find appropriate one so there is some risk that values are not optimal for SJW etc. But I have proposed this setup as the most portable for Linux kernel in LinCAN and then offered it to SocketCAN designers when they discussed how to parametrize setup and it works well for something like 15 years for all CAN and future added CAN FD controllers... There are specific cases where exact qualified setting for specific controller is required by carmakers, then alternative setting by exact raw parameters by application makes sense. Linux kernel allows to read parameters ranges by netlink interface and use alternative setting in raw values in such cases.